lib/capistrano/s3/publisher.rb

```diff
@@ -110,7 +112,7 @@ def self.put_object(s3, bucket, target_path, path, file, only_gzip, extra_option
           mime_type = mime_type_for_file(base_name)
           options   = {
             :bucket => bucket,
-            :key    => File.join(target_path, path),
+            :key    => self.add_prefix(path, prefix: target_path),
```

Comment:
Yes, but it's how current code in this library styled. I think it will be good to separate style changes into another PR.